server/service/src/sync/translations/item_direction.rs

```diff
@@ -0,0 +1,92 @@
+use repository::{ItemDirectionRow, ItemDirectionRowDelete, StorageConnection, SyncBufferRow};
+
+use serde::Deserialize;
+
+use crate::sync::translations::item::ItemTranslation;
+
+use super::{PullTranslateResult, SyncTranslation};
+
+#[allow(non_snake_case)]
+#[derive(Deserialize)]
+pub struct LegacyItemDirectionRow {
+    ID: String,
+    item_ID: String,
+    directions: String,
+    priority: i64,
+}
+
+// Needs to be added to all_translators()
+pub(crate) fn boxed() -> Box<dyn SyncTranslation> {
+    Box::new(ItemDirectionTranslation)
+}
+
+pub(super) struct ItemDirectionTranslation;
+impl SyncTranslation for ItemDirectionTranslation {
+    fn table_name(&self) -> &str {
+        "item_direction"
+    }
+
+    fn pull_dependencies(&self) -> Vec<&str> {
+        vec![ItemTranslation.table_name()]
+    }
+
+    fn try_translate_from_upsert_sync_record(
+        &self,
+        _: &StorageConnection,
+        sync_record: &SyncBufferRow,
+    ) -> Result<PullTranslateResult, anyhow::Error> {
+        let data = serde_json::from_str::<LegacyItemDirectionRow>(&sync_record.data)?;
+
+        let result = ItemDirectionRow {
+            id: data.ID,
+            item_link_id: data.item_ID,
```

Comment:
Is it safe to assume an `item_id` is also an `item_link_id`? Feels like we might need to look up the `item_id` based on this, but then I always get confused with `*_link` tables...

Reply:
The item link tables are populated with `item.id == item_link.id == item.item_link_id` so it should be fine I think!